local_planner/src/nodes/common.cpp

```diff
@@ -54,27 +54,42 @@ PolarPoint cartesianToPolar(double x, double y, double z,
 Eigen::Vector2i polarToHistogramIndex(const PolarPoint& p_pol, int res) {
   // TODO change logic here, as 0,0 are valid index
   Eigen::Vector2i ev2(0, 0);
-  float e = p_pol.e;
-  float z = p_pol.z;
-  if (res <= 0.f || e < -90.f || e > 90.f || z < -180.f || z > 180.f) {
-    return ev2;
+  PolarPoint p_wrapped = wrapPolar(p_pol);
+  // elevation angle to y-axis histogram index
+  if (p_wrapped.e == 90.0f) {
```

Comment:
I think this piece of logic doesn't make sense. I believe it was a hack to make things work with integers, I can't see what good it does here. Any opinions @mrivi ?

Reply:
Actually, after doing some math with pen and paper I see the point now. This is to make sure that the floor() will eventually return a valid index, even if the value is exactly on the upper edge of the last bin. I suggest however, to set it to 89.9f instead, to make it a bit more clear.

Reply:
this case is covered in the support function wrapPolar now.